### ISSUE TITLE
Crash To Desktop

### ISSUE THREAD

**Issue by amoper** [LoneGazebo/Community-Patch-DLL#1830]
Bug Report Template

Mod Version (i.e Date - (4/23b)):
4/23

1. Mod List (if using standard CPP set, leave blank):
The only addition is the In-Game Editor

2. Type of error (i.e. crash, interface bug, AI quirk): 
The game crashes on the Netherlands turn to the desktop. In the logs I see no error, but it's worth mentioning that I had this issue once before which I solved by deleting the civ the game crashed on. After that, the game continued normally. This time, though, I think I'm seeing a pattern where it happens occasionally when I'm at war with a civ. 

3. Steps to reproduce:
All you should have to do is load up the game, and press end turn. Once it gets to the Netherlands turn it should crash to desktop.

4. Additional information:
I think it's worth mentioning here that I have all the DLC needed, and this was done on a clean installation of W10. So, there is no possibility of any leftover mod files or anything (that I know of). Regardless, the only mods I've ever used anyway were CPP and In game editor.
---------------------------
Supporting information:

Please note that you can attach .zip files to Issues by dragging-and-dropping them. If possible, zip up all supporting data and post that way.

1. Log files: Database.log and Lua.log needed
2. Minidump file (located in your Civ5.exe directory)
3. Screenshots (if needed)

![mod_folder](https://cloud.githubusercontent.com/assets/19496016/15439022/f85fd0a2-1e9c-11e6-9422-08c61b2e49b6.png)
[Logs.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/275298/Logs.zip)

[CvMiniDump.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/275299/CvMiniDump.zip)

**Comment by LoneGazebo** [comment]
Why did you put Events.db in your MODS folder? Also, you don't need 6a and 6b, just one or the other depending on whether or not you are using EUI. Also, make sure you deleted the LUA folder of CP and CBP.

G

**Comment by amoper** [comment]
I just forgot to move it. But, to be safe I did move it as well as delete 6a (since I don't use EUI). Also, I deleted both LUA folders as well as the cache, logs, and Moduserdata folders just in case.

The game still crashes at the same turn. So I'm including fresh logs and dump.
[Logs.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/275682/Logs.zip)
[CvMiniDump.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/275684/CvMiniDump.zip)

**Comment by Hyronymus77** [comment]
amoper, I have the same also playing as the Dutch. And I think I even narrowed it down further:
- Hovering over enemy city banners of a city with a military unit in the city in peace time works;
- Hovering over enemy city banners of a city with a military unit in the city in wartime causes CTD's.

Difference is that I do use the EU version from 4/23. Can you check though that hovering over an enemy city banners of a city with a military unit in the city during wartime triggers the problem for you a well?

**Comment by Hyronymus77** [comment]
Update after thinking and experimenting a bit more:

**No CTD**
- Declare war on an enemy;
- Select and activate a ranged attack unit (land or naval);
- Hover over an enemy city in the fog of war with a garrison (check this with IGE).

**Reproducable CTD**
- Declare war on an enemy;
- Select and activate a melee attack unit (land or naval);
- Hover over an enemy city with a garrison (check this with IGE).

**Oddity**
Using a melee attack unit as descibed above early in the game causes no problems. Where does it go wrong? I rule out promotions as naval and land units obtain different promotions. Even promotions obtained from Wonders (Zeus, Himeji) cause no problem as I tried adding "n00b" units with IGE.

**Backlink**
Seems awkwardly similar to whoward69's two cents a this bug report: https://github.com/LoneGazebo/Community-Patch-DLL/issues/1746

**Setup**
Using the exact setup as the OP except for EUI which I do use.

**Comment by amoper** [comment]
I apologize I was at a wedding :)

No, hovering over the banners does not cause a crash for me. It is strictly the next turn sequence. To be honest, beyond being at war with them I can't really narrow it down. The fact that we are at war may itself be a coincidence :(

**Comment by LoneGazebo** [comment]
Try latest beta.

**Comment by amoper** [comment]
I'm sorry again for the late reply. I'm in the middle of registering for classes and internships at the moment so time is not my friend. I will try the latest beta this weekend (or whatever the latest happens to be at the time) and get back to you.

However, it is worth mentioning, I think, that this has happened before on other betas where I have to delete a civ to continue. Granted, whatever the issue is it could be solved with the newest one so I will test it out first as I realize you're not a wizard.

**Comment by Hyronymus77** [comment]
Could it be that the game crashes when a unit is moving through a city to another destination, ends its turns inside a city, gets bombarded and then causes the crash? As it is moving on it is not garrisoned but perhaps the code gets stuck there?

Garrisoned unit == unit inside city
Unit inside city != garrisoned unit

**Comment by amoper** [comment]
I had a little bit of down-time today so I tested a bit by deleting all my cargo ships (by going to war with everyone, lol), stopping all my units, and making sure no one moved from my side on that 'end turn' sequence. As for others, I am discounting that Japan attacked me, though, (since I needed to declare war to get rid of my cargo ships) since that didn't originally happen and the only one I was at war with originally (Netherlands) never attacked me or had a unit enter my territory on that turn.

So as far as I can tell no one is moving through the city. I'll attach the save file, though. Perhaps it's just me? I guess someone else can run it and see what happens.

I'll still check the latest version this weekend

[Nebuchadnezzar II_0331 AD-1911.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/287388/Nebuchadnezzar.II_0331.AD-1911.zip)
